### PR TITLE
Ensure git chart source before comparing mirror

### LINKED ISSUE #2026: Helm operator segfault observed with 0.9.0
**Description**
After deploying Flux 1.12.2 and Helm operator 0.9.0 a segfault was observed in helm-operator. Pod went into CrashLoopBackOff.

**To Reproduce**
Unsure. We updated flux and helm-op to the above versions in our dev cluster.

**Logs**

```
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator E0508 23:29:31.571287       7 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /go/src/github.com/weaveworks/flux/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /go/src/github.com/weaveworks/flux/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /go/src/github.com/weaveworks/flux/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /usr/local/go/src/runtime/asm_amd64.s:573
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /usr/local/go/src/runtime/panic.go:502
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /usr/local/go/src/runtime/panic.go:63
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /usr/local/go/src/runtime/signal_unix.go:388
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /go/src/github.com/weaveworks/flux/integrations/helm/chartsync/chartsync.go:266
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /go/src/github.com/weaveworks/flux/integrations/helm/chartsync/chartsync.go:166
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /go/src/github.com/weaveworks/flux/integrations/helm/chartsync/chartsync.go:166
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator /usr/local/go/src/runtime/asm_amd64.s:2361
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator panic: runtime error: invalid memory address or nil pointer dereference [recovered]
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator 	panic: runtime error: invalid memory address or nil pointer dereference
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1255973]
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator goroutine 54 [running]:
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator github.com/weaveworks/flux/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator 	/go/src/github.com/weaveworks/flux/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator panic(0x13cfbc0, 0x1ff5f10)
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator 	/usr/local/go/src/runtime/panic.go:502 +0x229
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator github.com/weaveworks/flux/integrations/helm/chartsync.mirrorName(...)
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator 	/go/src/github.com/weaveworks/flux/integrations/helm/chartsync/chartsync.go:266
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator github.com/weaveworks/flux/integrations/helm/chartsync.(*ChartChangeSync).getCustomResourcesForMirror(0xc4203aa5a0, 0xc420080000, 0x6c, 0x0, 0x0, 0x0, 0x0, 0x0)
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator 	/go/src/github.com/weaveworks/flux/integrations/helm/chartsync/chartsync.go:456 +0xf3
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator github.com/weaveworks/flux/integrations/helm/chartsync.(*ChartChangeSync).Run.func1(0xc4203aa5a0, 0xc4205ee2a0, 0xc420640060)
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator 	/go/src/github.com/weaveworks/flux/integrations/helm/chartsync/chartsync.go:166 +0x119
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator created by github.com/weaveworks/flux/integrations/helm/chartsync.(*ChartChangeSync).Run
default flux-helm-operator-675786ffd5-6br4l flux-helm-operator 	/go/src/github.com/weaveworks/flux/integrations/helm/chartsync/chartsync.go:155 +0xf8
```

### PR DESCRIPTION
**Pull request by hiddeco**
Fixes #2026

### CODE REVIEW

File: test/e2e/e2e-kind.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ chmod +x kubectl && \
 sudo mv kubectl /usr/local/bin/
 
 echo ">>> Building sigs.k8s.io/kind"
-go get -u sigs.k8s.io/kind
+GO111MODULE="on" go get -u sigs.k8s.io/kind
```
Comment (squaremo) [inline test/e2e/e2e-kind.sh]: Is this related?

Comment (hiddeco) [reply test/e2e/e2e-kind.sh]: Not related but without it I won't be able to merge.

Comment (hiddeco) [reply test/e2e/e2e-kind.sh]: I can move it to a separate PR though.

Comment (squaremo) [reply test/e2e/e2e-kind.sh]: Is it to avoid a conflict?

Comment (squaremo) [reply test/e2e/e2e-kind.sh]: (don't worry about moving to another PR)

Comment (squaremo) [reply test/e2e/e2e-kind.sh]: > Is it to avoid a conflict?

Ah no I see, an _externality_

Comment (hiddeco) [reply test/e2e/e2e-kind.sh]: Yeah, CircleCI is having a hard time anyway as it now seems to fail on the `ssh-keyscan` of bitbucket...

Comment (stefanprodan) [reply test/e2e/e2e-kind.sh]: @hiddeco you need to update go to 1.12.5 in e2e-golang.sh for go mod to work

**Pull request merged**